trick_source/sim_services/ScheduledJobQueue/ScheduledJobQueue.cpp

```diff
@@ -46,72 +71,24 @@ Trick::ScheduledJobQueue::~ScheduledJobQueue( ) {
 */
 int Trick::ScheduledJobQueue::push( JobData * new_job ) {
 
-    unsigned int ii , jj ;
-
     /* Allocate additional memory for the additional job in the queue */
-    JobData ** new_list = (JobData **)calloc( list_size + 1 , sizeof(JobData *)) ;
-
-    new_job->set_handled(true) ;
-
-    /* Find the correct insertion spot in the queue by comparing
-       the job_class, the phase, the sim_object id, and the job_id in that order. */
-    /* While searching for the correct insertion spot, copy all jobs that precede
-       the incoming job to the newly allocated queue space. */
-    for ( ii = jj = 0 ; ii < list_size ; ii++ ) {
-        if ( list[ii]->job_class == new_job->job_class ) {
-            if ( list[ii]->phase == new_job->phase ) {
-                if ( list[ii]->sim_object_id ==  new_job->sim_object_id ) {
-                    if ( list[ii]->id <= new_job->id ) {
-                        new_list[jj++] = list[ii] ;
-                    } else {
-                        new_list[jj++] = new_job ;
-                        break ;
-                    }
-                } else if ( list[ii]->sim_object_id < new_job->sim_object_id ) {
-                    new_list[jj++] = list[ii] ;
-                } else {
-                    new_list[jj++] = new_job ;
-                    break ;
-                }
-            } else if ( list[ii]->phase < new_job->phase ) {
-                new_list[jj++] = list[ii] ;
-            } else {
-                new_list[jj++] = new_job ;
-                break ;
-            }
-        } else if ( list[ii]->job_class < new_job->job_class ) {
-            new_list[jj++] = list[ii] ;
-        } else {
-            new_list[jj++] = new_job ;
-            break ;
-        }
+    JobData ** new_list = (JobData **)realloc(list, (list_size + 1) * sizeof(JobData *)) ;
+    if (!new_list) {
+        abort();
```

Comment:
This could be replaced with some other logic, but the old code wasn't handling allocation failure at all.